nasa-gibs/worldview
# WV-2638: Dynamic Data Visualization -updated

## Description

This is the implementation of the UI and further functionality for the dynamic data visualization layers that we are utilizing through the FIRMS API. This adds some layers with preset RGB band combinations and also a landsat customizable layer. This customizable layer allows for users to select custom RGB combinations. 

This imagery follows orbit tracks that can be added through the associated layers settings. In the instructions below I've added a URL with a good starting point. But if you want to find imagery on other dates, simply locate the orbit tracks on the date of interest (United States Only). 

I have also built out some tool components that can be used by developers to help debug and test features as they are built. This mode can only be engaged by setting the `tileImageTestMode` state variable in the `mapUI.js` file to true. 

## To Do

- [x] Add zot for under zoom levels
- [x] Add associated layer name in sidebar
- [x] Complete sentinel presets
- [x] Dynamic Wavelength/Bandwidth/Resolution Attributes
- [x] Update layer names (in [this PR](https://github.com/nasa-gibs/worldview/pull/4522))
- [x] Add layer previews (in [this PR](https://github.com/nasa-gibs/worldview/pull/4522))
- [x] Disable Snapshots for DDV Layers
- [x] Fix overzoom indicator zot
- [x] Fix naming conventions
- [x] Fix HLS_Natural_Color_Landsat showing up as Sentinel 2 instead of Landsat
- [x] Add Sentinel layer for 6-5-4 Vegetative Analysis False Color?
- [x] Limit Landsat Band Options to 1-9

## How To Test

1. `git checkout dynamic-data-viz-exp`
2. `npm ci`
3. `npm run build`
4. `npm run watch`
5. For simplicity, use this [URL](http://localhost:3000/?v=-77.27333274057173,38.77605197284668,-76.83722700850396,38.97116386026658&l=Reference_Labels_15m,Reference_Features_15m,MODIS_Terra_CorrectedReflectance_TrueColor&lg=true&t=2023-04-17-T12%3A21%3A36Z) as a starting point. This will place you over Washington DC on April 17th where any of the new DDV layers will be visible. 
6. Go into the layer picker and add and remove each preset DDV layer (HLS Natural Color, HLS Color Infrared, HLS False Color Vegetation, HLS False Color Urban, HLS Shortwave Infrared). For each layer, wait for the tiles to load and add the associated orbit track in the layer options. 
7. Add the 'HLS Customizable Landsat' layer. 
8. Open the layer options and verify that the current band selection defaults to RED: B07 GREEN: B05 BLUE: B04
9. Click on the customize button within the layer settings and confirm the new customization menu opens. 
10. Create a few custom combinations by changing the values of R, G & B in the bottom selection and then hitting the 'Confirm Band Selection' button. NOTE: Some combinations do not exist! Especially when using the B8A band and higher. We will have to figure out a way to tell what bands can actually be used in any combination. This was _not_ a problem before adding bands above B08.
11. This should close the customization menu and you should shortly be able to visualize your changes on the map. 
12. After testing a few custom combinations, try selecting and confirming the preset values from the customizable menu. 
13. Verify that the preset preview images look similar to the visualization you see on the map.
